core/src/main/java/org/mule/runtime/core/api/context/MuleContextBuilder.java

```diff
@@ -80,4 +83,17 @@ static MuleContextBuilder builder(ArtifactType artifactType) {
    * @since 4.5.0
    */
   void setArtifactCoordinates(ArtifactCoordinates artifactCoordinates);
+
+  /**
+   * @return the {@link Registry} of the built {@link MuleContext}.
+   *         <p>
+   *         This method must be called after {@link #buildMuleContext()}, otherwise it will always return
+   *         {@link Optional#empty()}.
+   *
+   * @since 4.6.0
+   */
+  default Optional<Registry> getRegistry() {
```

Comment:
the builder should most definitively not be capable of returning the registry, and there should be no expectation for the registry to exist before the build() mthod is invoked